application/src/main/java/org/opentripplanner/street/model/vertex/VertexLabel.java

```diff
@@ -59,6 +59,20 @@ public String toString() {
     }
   }
 
+  /**
+   * A vertex label for a split vertex generated for a node on a way.
+   * Must be similar to the parent vertex label so that tests can
+   * be written sensibly.
+   */
+  record NodeOnWayLabel(long nodeId, long wayId) implements VertexLabel {
+    private static final String TEMPLATE = "osm:node:%s:way:%s";
+
```

Comment:
I accidentally noticed that 'way' can be a relation as well. Is there a bug in logic, or is this just a minor naming issue?

Reply:
This is just a naming issue, I think it happens elsewhere as well.